services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationController.kt

```diff
@@ -98,22 +108,25 @@ class ApplicationController(
                 ),
             ]
     )
-    fun create(@ValidApplication @RequestBody application: Application): Application {
+    fun create(
+        @RequestParam("luo-hanke") generateHanke: Boolean = false,
+        @ValidApplication @RequestBody application: Application
+    ): Application {
+        logger.info { "Creating new application, param luo-hanke: $generateHanke" }
         val userId = currentUserId()
-        val hankeTunnus = application.hankeTunnus
-        val hankeId = hankeService.getHankeId(hankeTunnus)
 
-        if (
-            hankeId == null ||
-                !permissionService.hasPermission(hankeId, userId, PermissionCode.EDIT_APPLICATIONS)
-        ) {
-            throw HankeNotFoundException(hankeTunnus)
+        if (generateHanke) {
```

Comment:
This function two completely different paths depending on the value of the `luo-hanke` parameter. I think it would make more sense to have these as different methods with different URLs. Using a query parameter to separate the two flows feels hacky.

Reply:
There are now to different endpoints to create and create with hanke generation.